provisionerd/provisionerd.go

```diff
@@ -205,8 +211,8 @@ func (p *Server) connect(ctx context.Context) {
 		if p.isClosed() {
 			return
 		}
-		ticker := time.NewTicker(p.opts.PollInterval)
-		defer ticker.Stop()
+		timer := time.NewTimer(p.opts.JobPollInterval)
```

Comment:
Wouldn't it be better to only apply this delay if it failed to acquire a job? This means we will acquire them as quick as possible without delays if there are a lot of jobs queued, but during periods of low jobs we will only attempt to acquire a job every 5 seconds.

Reply:
I guess in theory yeah, but it's currently architected in a way where this is still ticking while the provisioner is running a job, so it'd require a bit of a refactor for this to work.

Reply:
couldn't you just do something like:
```go
for {
  ok :=acquireJob()
  if !ok {
    select {
      case <-ctx.Done():
        return ctx.Err()
      case <-time.After(5 * time.Second):
  }
}
```

or something? time.After's only leak if they haven't fired yet, so since the interval is small and the only time this for loop will exit is if the app is shutting down it seems fine here